File: 프로그래머스/lv2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        int w = 3;
        int total = brown + yellow;
        
        find:
        while(true) {
            for(int h = 3;h <= w;h++) {
                if(h * w != total) continue;
                else {
                    int sum = 2 * w + ((h - 2) * 2);
                    if(total - sum == yellow) {
                        answer[0] = w;
                        answer[1] = h;
                        break find;
                    } 
                }
            }
            w += 1;
        }
        
        return answer;
    }
}

File: 프로그래머스/Lv.2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    
    let h = 3;
    const total = brown + yellow;
    
    while(true) {
        let flag = false;
        for(let w = 3;w <= Math.floor(total / h);w++) {
            if(h * w === total) {
                if((h - 2) * (w - 2) === yellow) {
                    flag = true;
                    answer[0] = w;
                    answer[1] = h;
                    break;
                }
            }
        }
        if(flag) break;
        else h++;
    }
    
    return answer;
}